pdc_client/__init__.py

```diff
@@ -161,7 +258,13 @@ def decode(req):
 
         content_type = "application/json"
         encode = json.dumps
-        self.client = beanbag.BeanBag(url, session=self.session, fmt=(content_type, encode, decode))
+        # In get_paged function it returns an interator, so user can get each pages quickly in plugins.
+        # The original call function of BeanBag returns the list.
+        # Keep the current codes and options in codes. Just init MultiPageBeanBag when is_multi is true.
+        if is_multi:
+            self.client = MultiPageBeanBag(url, session=self.session, fmt=(content_type, encode, decode))
```

Comment:
Why not use just the new class without having the extra option and have `yield` there?

Reply:
@hluk Is that ok to merge? or is there anything which still  need change from your point of view?

Reply:
@hluk any updates?

Reply:
Following code seems extremely slow (for me it takes ~2 seconds to fetch single page).

```python
import pdc_client
c = pdc_client.PDCClient('qa', is_multi=True)
c['releases']()
```

I.e. doing multiple requests could take forever. I suggest overriding `page_size=-1` ... but that would actually mean that the new class is not needed.

We should probably talk about this tomorrow over phone/video.

Reply:
Based on this issue, the requirement is about  the logic like this. So this class is still needed.
And also the test in your case is just for less data. If for large data, get data once a time maybe have issue about the http request.
Also even in this case you can also try like this:

```
import pdc_client
c = pdc_client.PDCClient('qa')
c['releases'](page_size="-1")
```
Let's talk about this by phone/video then.